### PR TITLE
Docker and CI instructions in README

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
#### Purpose/implementation

Adding instructions for adding dependencies to the Docker container and will add instructions around CI in an upcoming commit.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
@cgreene what do you think about adding an issue template for Docker assistance?

File: README.md [hunk 1]
```diff
@@ -81,10 +81,41 @@ We noticed ControlFreeC does not properly handle aneuploidy well for a subset of
...
 ## How to Add an Analysis
 
-Users performing analyses, should always refer to the symlinks in the `data/` directory and not files within the release folder, as an updated release may be produced before a publication is prepared.
+Users performing analyses, should **always** refer to the symlinks in the `data/` directory and not files within the release folder, as an updated release may be produced before a publication is prepared.
+
+### Docker Image
+
+We build our project Docker image from a versioned [`tidyverse`](https://hub.docker.com/r/rocker/tidyverse) image from the [Rocker Project](https://www.rocker-project.org/) (v3.6.0).
+
+To add dependencies that are required to your analysis to the project Docker image, you
```
Comment (cgreene) [inline README.md]:
```suggestion
To add dependencies that are required for your analysis to the project Docker image, you
```

Comment (cgreene) [reply README.md]: may have a bonus line break here too

File: README.md [hunk 2]
```diff
@@ -81,10 +81,41 @@ We noticed ControlFreeC does not properly handle aneuploidy well for a subset of
...
+must alter the project [`Dockerfile`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/Dockerfile).
+Note: R packages installed on this image will be installed from an [MRAN snapshot](https://mran.microsoft.com/documents/rro/reproducibility#reproducibility) corresponding to the last day that R 3.6.0 was the most recent release ([ref](https://hub.docker.com/r/rocker/tidyverse)).
+
+If you need assistance adding a dependency to the Dockerfile, [file a new issue on this repository](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/new) to request help.
+
+#### Development in the Project Docker Container
+
+The most recent version of the project Docker image, which is pushed to Docker Hub after a pull request gets merged into the master branch, can be obtain via the command line with:
```
Comment (cgreene) [inline README.md]:
```suggestion
The most recent version of the project Docker image, which is pushed to Docker Hub after a pull request gets merged into the master branch, can be obtained via the command line with:
```

**Comment by jaclyn-taroni** [comment]
I think the best way to test this is to have the open analysis PRs follow these directions and tweak as necessary.

**Comment by cgreene** [comment]
@jaclyn-taroni agree. Would request review, after that passes merge, and then make changes that come up as this process happens.

File: README.md [hunk 3]
```diff
@@ -144,4 +174,35 @@ Files that are intermediate, which means that they are useful within an analysis
 
 ### Continuous Integration (CI)
 
-### Adding Analyses to CI
+We continuous integration (CI) to ensure that all analysis code will execute and that the project Docker image will build if there are any changes introduced to the [`Dockerfile`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/Dockerfile).
```
Comment (cgreene) [inline README.md]:
```suggestion
We use continuous integration (CI) to ensure that the project Docker image will build if there are any changes introduced to the [`Dockerfile`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/Dockerfile) and that all analysis code will execute.
```

File: README.md [hunk 4]
```diff
@@ -144,4 +174,35 @@ Files that are intermediate, which means that they are useful within an analysis
...
+
+We have put together files specifically for the purpose of CI that contain the full features in a data file, but only for a small subset of samples.
```
Comment (cgreene) [inline README.md]:
```suggestion
We have put together data files specifically for the purpose of CI that contain all of the features of the full data files for only a small subset of samples. 
```

File: README.md [hunk 5]
```diff
@@ -144,4 +174,35 @@ Files that are intermediate, which means that they are useful within an analysis
 
 ### Continuous Integration (CI)
 
-### Adding Analyses to CI
+We continuous integration (CI) to ensure that all analysis code will execute and that the project Docker image will build if there are any changes introduced to the [`Dockerfile`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/Dockerfile).
+
+We have put together files specifically for the purpose of CI that contain the full features in a data file, but only for a small subset of samples. 
+(You can see how this was done by viewing [this notebook](https://alexslemonade.github.io/OpenPBTA-analysis/analyses/create-subset-files/01-create_subset_files.nb.html).)
```
Comment (cgreene) [inline README.md]:
```suggestion
You can see how this was done by viewing [this notebook](https://alexslemonade.github.io/OpenPBTA-analysis/analyses/create-subset-files/01-create_subset_files.nb.html).
```

File: README.md [hunk 6]
```diff
@@ -144,4 +174,35 @@ Files that are intermediate, which means that they are useful within an analysis
...
+We use the subset files to cut down on the computational resources and time required for testing.
+Provided that your analytical code points to the symlinks in the `data/` directory per [the instructions above](#how-to-add-an-analysis), adding the analysis to the CI will run on these subset files (see below).
```
Comment (cgreene) [inline README.md]:
```suggestion
Provided that your analytical code points to the symlinks in the `data/` directory per [the instructions above](#how-to-add-an-analysis), adding the analysis to the CI (see below) will run your analysis on this subset of the data.
```

File: README.md [hunk 7]
```diff
@@ -144,4 +174,35 @@ Files that are intermediate, which means that they are useful within an analysis
...
+Note that if you hardcode sample names in your analytical code and those samples are not present in the subset files, this may cause errors.
```
Comment (cgreene) [inline README.md]:
```suggestion
Do not hardcode sample names in your analytical code: there is no guarantee that those samples will be present in the subset files.
```

File: README.md [hunk 8]
```diff
@@ -144,4 +174,35 @@ Files that are intermediate, which means that they are useful within an analysis
...
+
+#### Adding Analyses to CI
+
+For an analysis to be run in CI, it must be added to the Circle CI configuration file, [`.circleci/config.yml`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/.circleci/config.yml). 
+A new analysis should be added as the last step of the `run_analyses` job.
```
Comment (cgreene) [inline README.md]:
```suggestion
A new analysis should be added as the last step of the `run_analyses` section.
```

**Pull request merged**